### PR TITLE
docs/enterprise: add the Thales HSM install guide to the public repo

### PR DESCRIPTION
**Pull request by bobg**

### CODE REVIEW

**Comment by bobg** [comment]
PTAL

File: docs/enterprise/thales-install-guide.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Using the Thales nShield Connect Hardware Security Module with Chain Core
```
Comment (ryandotsmith) [inline docs/enterprise/thales-install-guide.md]: Perhaps we should reduce the number of words in the h1. All of our other doc pages use a single line for the h1.

**Comment by bobg** [comment]
Updated to describe presigned firmware; PTAL @kr and @ryandotsmith

File: docs/enterprise/thales-install-guide.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+   - Create an administrator cardset and an operator cardset
+
+   These steps are complex and involve policy decisions and other
+   choices.  Refer to the Thales documentation for details.  Note:
+   Chain has tested this using 64-bit Ubuntu Linux on the client host.
+
+-  On the client host, use the “nfast” utilities from Thales to create
+   a _user-data signing key_ and a signed copy of the userdata file.
+   - `generatekey --batch seeinteg plainname=xprvseemoduledevusk`
+   - `tct2 --sign-and-pack -k xprvseemoduledevusk --machine-key=68bcec164114318f31b2e353bef9e8b1ce67872e -o userdata.sar -i userdata.bin`
```
Comment (kr) [inline docs/enterprise/thales-install-guide.md]: Somewhere we should note that this hash identifies a **development** key and should not be trusted for production.

**Comment by kr** [comment]
LGTM

**Pull request merged**